tutor/commands/upgrade/compose.py

```diff
@@ -158,6 +158,34 @@ def upgrade_from_olive(context: click.Context, config: Config) -> None:
     upgrade_mongodb(context, config, "4.2.17", "4.2")
     upgrade_mongodb(context, config, "4.4.22", "4.4")
 
+    upgrade_to_redwood_onwards, new_mysql_docker_image = (
+        common_upgrade.verify_tutor_version_for_mysql_upgrade(config)
+    )
+
+    if not upgrade_to_redwood_onwards:
+        return
+
+    # Revert the MySQL image first to build data dictionary on v8.1
+    old_mysql_docker_image = "docker.io/mysql:8.1.0"
+    click.echo(fmt.title(f"Upgrading MySQL to v{new_mysql_docker_image.split(':')[1]}"))
+    config["DOCKER_IMAGE_MYSQL"] = old_mysql_docker_image
+    # Note that the DOCKER_IMAGE_MYSQL value is never saved, because we only save the
+    # environment, not the configuration.
```

Comment:
curious: So this only keep the change in that terminal environment and not to file, correct?

Reply:
Correct.

Similar to how it is being done in the MongoDB upgrade.

https://github.com/overhangio/tutor/blob/3143335d69a7f9b012551d14838ea81101101551/tutor/commands/upgrade/compose.py#L187-L190